packages/arb-token-bridge-ui/src/hooks/useChainId.ts

```diff
@@ -5,11 +5,20 @@ export function useChainId({ provider }: { provider: providers.Provider }) {
   const [chainId, setChainId] = useState<number | undefined>(undefined)
 
   useEffect(() => {
+    let cancelled = false
+
     async function updateChainId() {
```

Comment:
We need to use the latest `chainid`, otherwise we run into issues where `l1` `l2` and `isDepositMode` change almost simultaneously, trigger the update and overwrite the latest.

Reply:
Did not understand this part, where exactly is the impact of this? Orbit chainId is not getting loaded if this change is not made? 🤔